src/txn/txn_rollback_to_stable.c

```diff
@@ -44,6 +44,20 @@ __rollback_abort_newer_update(WT_SESSION_IMPL *session, WT_UPDATE *first_upd,
               upd->txnid, __wt_timestamp_to_string(upd->durable_ts, ts_string[0]),
               __wt_timestamp_to_string(rollback_timestamp, ts_string[1]));
 
+            upd->txnid = WT_TXN_ABORTED;
+            WT_STAT_CONN_INCR(session, txn_rts_upd_aborted);
+            upd->durable_ts = upd->start_ts = WT_TS_NONE;
+        } else if (upd->prepare_state == WT_PREPARE_INPROGRESS ||
+          upd->prepare_state == WT_PREPARE_LOCKED) {
```

Comment:
This `else` branch and the previous clause `else if (rollback_timestamp < upd->durable_ts)` appear to be identical except for the `WT_ASSERT`.  Can we refactor so there isn't repeated code?

Reply:
Fixed.